iothub/device/src/Transport/ProtocolRoutingDelegatingHandler.cs

```diff
@@ -31,6 +31,63 @@ internal class ProtocolRoutingDelegatingHandler : DefaultDelegatingHandler
         {
         }
 
+        public override async Task OpenAsync(TimeoutHelper timeoutHelper)
+        {
+            try
+            {
+                if (Logging.IsEnabled) Logging.Enter(this, timeoutHelper, $"{nameof(ProtocolRoutingDelegatingHandler)}.{nameof(OpenAsync)}");
+                
+                bool gain = await _handlerLock.WaitAsync(timeoutHelper.RemainingTime()).ConfigureAwait(false);
```

Comment:
Same as above: isn't this the same code as the OpenAsync with cancellation token? Can you extract a method and avoid code duplication?

Reply:
only disposing part is common, leaving it as it is. Extracting just disposing part does not make much of sense